### PR TITLE
Add trigger_sync_run_and_wait_for_completion flow

### LINKED ISSUE #10: Add `trigger_hightouch_sync_and_wait_for_completion` flow
Create a flow that triggers a Hightouch sync and waits for it to reach a terminal state. 

The flow should accept:
 - a `HightouchCredentials` block
 - a sync ID
 - a flag to specify if a full resync should be performed
 - an optional number of seconds to wait for completion
 - an optional poll frequency
 
The flow should return the final details of the final sync run when successful. The flow should raise if the sync run was unsuccessful for any reason.

### PR DESCRIPTION
**Pull request by ahuang11**
## Summary

Example:
```python
from prefect import flow

from prefect_hightouch import HightouchCredentials
from prefect_hightouch.syncs import trigger_sync_run_and_wait_for_completion

@flow
def sync_flow():
    hightouch_credentials = HightouchCredentials.load("hightouch-token")
    sync_metadata = trigger_sync_run_and_wait_for_completion(
        hightouch_credentials=hightouch_credentials,
        sync_id=58784,
        full_resync=True
    )
    return sync_metadata
```
Outputs:
image

## Relevant Issue(s)

Closes https://github.com/PrefectHQ/prefect-hightouch/issues/10

## Checklist
- [x] Summarized PR's changes in [CHANGELOG.md](https://github.com/PrefectHQ/prefect-hightouch/blob/main/CHANGELOG.md)
- [x] add test
- [x] Add pragma no cover to ellipses (or maybe replace with pass?) since:
```
prefect_hightouch/syncs/generated.py            29      5      0      0    83%   76, 113, 155, 199, 235
```

### CODE REVIEW

**Comment by ahuang11** [comment]
Wondering if I should rename `sync_run_wait_for_completion` to `wait_for_sync_run_completion`
image

**Comment by desertaxle** [comment]
> Wondering if I should rename `sync_run_wait_for_completion` to `wait_for_sync_run_completion`

I vote for `wait_for_sync_run_completion`

File: prefect_hightouch/syncs/flows.py [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    HightouchSyncRunTimedOut,
+)
+from prefect_hightouch.syncs import get_sync, trigger_run
+
+
+@flow
+async def trigger_sync_run_and_wait_for_completion(
+    hightouch_credentials: HightouchCredentials,
+    sync_id: str,
+    full_resync: bool = True,
```
Comment (desertaxle) [inline prefect_hightouch/syncs/flows.py]: The API defaults this to false, so I think we should too.
```suggestion
    full_resync: bool = False,
```

File: prefect_hightouch/syncs/flows.py [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+        poll_frequency_seconds=poll_frequency_seconds,
+    )
+
+    if sync_status == SyncStatus.SUCCESS:
+        return sync_metadata
+    else:
+        raise TERMINAL_STATUS_EXCEPTIONS[sync_status](
+            f"Sync ({sync_metadata.slug!r}, ID {sync_id!r}) "
+            f"was unsuccessful with {sync_status.value!r} status"
+        )
```
Comment (desertaxle) [inline prefect_hightouch/syncs/flows.py]: Can we use `.get()` here and raise a `HightouchSyncRunError` if we get a unexpected status back?

File: prefect_hightouch/exceptions.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
+"""
+This is a module containing exceptions used within prefect-hightouch.
+"""
+
+from prefect_hightouch.api_client.models import SyncStatus
+
+
+class HightouchSyncRunError(RuntimeError):
```
Comment (desertaxle) [inline prefect_hightouch/exceptions.py:8]: Big fan of this exception structure!

File: prefect_hightouch/syncs/flows.py [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+
+        sync_flow()
+        ```
+    """
+    json_body = TriggerRunInput(full_resync=full_resync)
+    sync_run_future = await trigger_run.submit(
+        hightouch_credentials=hightouch_credentials,
+        sync_id=sync_id,
+        json_body=json_body,
+    )
```
Comment (desertaxle) [inline prefect_hightouch/syncs/flows.py:108]: Can we add a log statement saying that we successfully triggered the sync and potentially give a URL to the Hightouch app to view progress?

**Review by desertaxle — CHANGES_REQUESTED** [review]
Looks good! Few small comments from me.

File: prefect_hightouch/exceptions.py (hunk 3 above)
Comment (ahuang11) [reply prefect_hightouch/exceptions.py:8]: :) Thanks, inspired by your work in dbt Cloud, Nate separating into exceptions module, and the Dan Bader's Python Tricks book for inheriting!

File: prefect_hightouch/syncs/flows.py (hunk 4 above)
Comment (ahuang11) [reply prefect_hightouch/syncs/flows.py:108]: I don't think I can rebuild it.

Example URL:
```
https://app.hightouch.com/myworkspace-abcdef/sources/1234/runs/123456
```

I have all the IDs, but I am not sure how I can get myworkspace-abcdef

Comment (ahuang11) [reply prefect_hightouch/syncs/flows.py:108]: I can provide:
`09:59:13.869 | INFO    | Flow run 'fervent-stingray' - Started sync '1234' run '123456'; open https://app.hightouch.com/ and append /sources/1234/runs/123456 to view results on webpage.`

Comment (desertaxle) [reply prefect_hightouch/syncs/flows.py:108]: I made #16 for tracking. We can find a solution to that in a separate PR.

**Review by desertaxle — DISMISSED** [review]

**Review by desertaxle — APPROVED** [review]

**Pull request merged**